lcls_tools/common/measurements/measurement.py

```diff
@@ -0,0 +1,16 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from pydantic import BaseModel, DirectoryPath
+
+
+class Measurement(BaseModel, ABC):
+    name: str
+    save_data: bool = True
+    save_location: Optional[DirectoryPath] = None
+
+    @abstractmethod
+    def measure(self) -> dict:
+        """ Implements a measurement and returns a dictionary with the results"""
+        pass
```

Comment:
Trying to think of what else besides metadata would be common to all measurements? I'm sure there's something.

Reply:
I imagine something will come up as we implement more measurements